### PR TITLE
feat: early redirect for image

### PR DESCRIPTION
**Pull request by preschian**
- [x] closes https://github.com/kodadot/nft-gallery/issues/4595

| before | after |
| --- | --- |
| Screenshot 2023-01-07 at 20 13 25 | Screenshot 2023-01-07 at 20 12 23 |

on already uploaded images to cf-images and hit the fetch cache will be faster (<1s). on a new image still takes > 1s

### CODE REVIEW

File: image/src/index.ts [hunk 1]
```diff
@@ -84,7 +73,24 @@ app.all('/ipfs/:cid', async (c) => {
...
 
     if (!response) {
-      // else, redirect to cf-images or render existing r2 object
+      const cfImage = `https://imagedelivery.net/${c.env.CF_IMAGE_ID}/${cid}/public`;
+      const currentImage = await fetch(cfImage, {
+        method: 'HEAD',
+        cf: {
+          cacheTtlByStatus: {
+            '200-299': CACHE_MONTH,
+            '404': 1,
```
Comment (vikiival) [inline image/src/index.ts:82]: Create also const for this for better readablility

File: image/src/index.ts [hunk 2]
```diff
@@ -84,7 +73,24 @@ app.all('/ipfs/:cid', async (c) => {
     }
 
     if (!response) {
-      // else, redirect to cf-images or render existing r2 object
+      const cfImage = `https://imagedelivery.net/${c.env.CF_IMAGE_ID}/${cid}/public`;
+      const currentImage = await fetch(cfImage, {
+        method: 'HEAD',
+        cf: {
+          cacheTtlByStatus: {
```
Comment (vikiival) [inline image/src/index.ts:80]: Move whole object to constants?

**Review by vikiival — CHANGES_REQUESTED** [review]
Very nice!
Love it 💕

Smol stuff for better readability

**Comment by yangwao** [comment]
wow, that's like 80% cut in time? 😍

**Review by vikiival — APPROVED** [review]
Very nice!

**Comment by vikiival** [comment]
Published

```
Uploaded image (2.71 sec)
Published image (1.64 sec)
  https://image.kodadot.workers.dev
Current Deployment ID: 0cd33751-e821-49d8-967b-f617056596ac
```

**Pull request merged**